src/garage/torch/algos/ddpg.py

```diff
@@ -0,0 +1,300 @@
+"""This modules creates a DDPG model in PyTorch."""
+from collections import deque
+
+from dowel import logger, tabular
+import numpy as np
+import torch
+
+from garage.np.algos.off_policy_rl_algorithm import OffPolicyRLAlgorithm
+
+
+class DDPG(OffPolicyRLAlgorithm):
+    """A DDPG model implemented with PyTorch.
+
+    DDPG, also known as Deep Deterministic Policy Gradient, uses actor-critic
+    method to optimize the policy and reward prediction. It uses a supervised
+    method to update the critic network and policy gradient to update the actor
+    network. And there are exploration strategy, replay buffer and target
+    networks involved to stabilize the training process.
+
+    Args:
+        env_spec (EnvSpec): Environment specification.
+        policy (garage.torch.policies.base.Policy): Policy.
+        target_policy (garage.torch.policies.base.Policy): Copy of policy.
+        qf (object): Q-value network.
+        target_qf (object): Copy of Q-value network.
+        replay_buffer (garage.replay_buffer.ReplayBuffer): Replay buffer.
+        n_epoch_cycles (int): Number of train_once calls per epoch.
```

Comment:
isn't this a parameter of the runner?

Reply:
Will remove.

Reply:
hmm can you confirm with @zequnyu and @naeioi ?

this seems strange to me. i was under the impression we refactored this logic out of the algorithms.

Reply:
If we want to remove `n_epoch_cycles` from algorithm, there is a way to access it from algorithm.train() via `runner.train_args`, but I feel it's too complicated. We had an issue in #736 however we haven't yet made the change.

Reply:
Oh, I think we refactored `n_epoch_cycles` back in, since it's only used by `OffPolicyAlgorithm`?

Reply:
It's also used in cem and cma_es. it's called `n_samples`.

Reply:
I am pretty sure these parameters were introduced back into the algorithms in a PR for local runner recently @ryanjulian .